app-modules/segment/src/Filament/Resources/SegmentResource/Pages/ListSegments.php

```diff
@@ -76,7 +77,24 @@ public function table(Table $table): Table
             ])
             ->actions([
                 EditAction::make(),
-                DeleteAction::make(),
+                DeleteAction::make()
+                    ->action(function (DeleteAction $action) {
+                        if ($action->getRecord()->campaigns()->exists()) {
```

Comment:
Let's do this check in the Policy of the Segment, returning a Deny response with the message.

We can then authorize the User here and return the notification with the message and cancel the action if it is denied.

Reply:
> Let's do this check in the Policy of the Segment, returning a Deny response with the message.
> 
> We can then authorize the User here and return the notification with the message and cancel the action if it is denied.

For notification message , DeleteAction will not visible once we provide the condition in Policy.

Reply:
You still need to do the `action` override code you had but capture the denial and usage the message from it in the notification.